openhands/integrations/bitbucket/bitbucket_service.py

```diff
@@ -385,6 +393,129 @@ async def create_pr(
         # Return the URL to the pull request
         return data.get('links', {}).get('html', {}).get('href', '')
 
+    async def get_microagents(self, repository: str) -> list[MicroagentResponse]:
+        """Fetch microagents from Bitbucket repository using Bitbucket API.
+
+        Args:
+            repository: Repository name in format 'workspace/repo_slug'
+
+        Returns:
+            List of microagents found in the repository (without content for performance)
+        """
+        microagents_path = self._determine_microagents_path(repository)
+        microagents = []
+
+        # Step 1: Get repository details - return early if not found
+        try:
+            repo_details = await self.get_repository_details_from_repo_name(repository)
+        except Exception as e:
+            raise UnknownException(
+                f'Error fetching repository details from Bitbucket: {str(e)}'
+            )
```

Comment:
`make_request` already raises the correct exception type, so we don't need to wrap it with UnknownException again

Reply:
Actually we don't need a try catch here since the exception will be raised, the provider handler can handle it

Reply:
Hi @malhotra5, I have updated the code based on your feedback. Thank you for your feedback!